system_tests/espresso_test.go

```diff
@@ -0,0 +1,282 @@
+	"github.com/offchainlabs/nitro/validator/valnode"
+)
+
+var (
+	validationPort = 54320
+	broadcastPort  = 9642
+)
+
+func espresso_block_txs_generators(t *testing.T, l2Info *BlockchainTestInfo) map[int][][]byte {
+	return map[int][][]byte{
+		5:  onlyMalformedTxs(t),
+		15: userTxs(t, l2Info),
+		25: func(t *testing.T) [][]byte {
+			// Contains malformed txes, valid transactions and invalid transactions
+			r := [][]byte{}
+			r = append(r, onlyMalformedTxs(t)...)
+			r = append(r, userTxs(t, l2Info)...)
+			return r
+		}(t),
+	}
+}
+
+func onlyMalformedTxs(t *testing.T) [][]byte {
+	return [][]byte{
+		{1, 2, 3},
+		{1, 2, 3},
+		{4, 5, 6},
+		{1, 2, 4},
+	}
+}
+
+// Two valid transactions and 2 invalid transactions with invalid nonces
+func userTxs(t *testing.T, l2Info *BlockchainTestInfo) [][]byte {
+	tx1 := l2Info.PrepareTx("Faucet", "Owner", 3e7, big.NewInt(1e16), nil)
+	tx1Bin, err := json.Marshal(tx1)
+	if err != nil {
+		panic(err)
+	}
+	tx2 := l2Info.PrepareTx("Owner", "Faucet", 3e7, big.NewInt(1e16), nil)
+	tx2Bin, err := json.Marshal(tx2)
+	if err != nil {
+		panic(err)
+	}
+	// 2 valid transactions here
+	return [][]byte{
+		tx1Bin,
+		tx1Bin,
+		tx2Bin,
+		tx2Bin,
+	}
+}
+
+func createMockHotShot(ctx context.Context, t *testing.T, l2Info *BlockchainTestInfo) (func(), int) {
+	httpmock.Activate()
+
+	httpmock.RegisterResponder(
+		"POST",
+		"http://127.0.0.1:50000/submit/submit",
+		func(req *http.Request) (*http.Response, error) {
+			log.Info("POST", "url", req.URL, "body", req.Body)
+			return httpmock.NewJsonResponse(200, nil)
+		},
+	)
+
+	httpmock.RegisterResponder(
+		"GET",
+		`=~http://127.0.0.1:50000/availability/header/(\d+)`,
+		func(req *http.Request) (*http.Response, error) {
+			log.Info("GET", "url", req.URL)
+			block := uint64(httpmock.MustGetSubmatchAsUint(req, 1))
+			// todo
+			header := espresso.Header{
+				TransactionsRoot: espresso.NmtRoot{Root: []byte{}},
+				Metadata: espresso.Metadata{
+					L1Head:    block,
+					Timestamp: uint64(time.Now().Unix()),
+				},
+			}
+			return httpmock.NewJsonResponse(200, header)
+		})
+
+	generators := espresso_block_txs_generators(t, l2Info)
+	dummyProof := func(n int) []byte {
+		r, _ := json.Marshal(map[int]int{0: n})
+		return r
+	}
+
+	httpmock.RegisterResponder(
+		"GET",
+		`=~http://127.0.0.1:50000/availability/block/(\d+)/namespace/100`,
+		func(req *http.Request) (*http.Response, error) {
+			txes := []espresso.Transaction{}
+			block := int(httpmock.MustGetSubmatchAsInt(req, 1))
+			data, ok := generators[block]
+			proof := dummyProof(block)
+			if block > 100 {
+				// make the debug message more cleaner
+				return httpmock.NewJsonResponse(404, 0)
+			}
+			log.Info("GET", "url", req.URL)
+			if !ok {
+				r := espresso.NamespaceResponse{
+					Proof:        (*json.RawMessage)(&proof), // todo
+					Transactions: &[]espresso.Transaction{},
+				}
+				return httpmock.NewJsonResponse(200, r)
+			}
+			for _, rawTx := range data {
```

Comment:
I think that generators (or whatever we end up using to generate txns) can do this wrappig